FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,270 @@
+       appDistributionSignInCompletion:(void (^)(NSError *_Nullable error))completion {
+  if (!configuration) {
+    // TODO: Handle when we cannot get configuration
+    @throw([NSException exceptionWithName:@"NotImplementedException"
+                                   reason:@"This code path is not implemented yet"
+                                 userInfo:nil]);
+    return;
+  }
+
+  NSString *redirectURL = [@"dev.firebase.appdistribution."
+      stringByAppendingString:[[[NSBundle mainBundle] bundleIdentifier]
+                                  stringByAppendingString:@":/launch"]];
+
+  OIDAuthorizationRequest *request = [[OIDAuthorizationRequest alloc]
+      initWithConfiguration:configuration
+                   clientId:kTesterAPIClientID
+                     scopes:@[ OIDScopeOpenID, OIDScopeProfile, kOIDScopeTesterAPI ]
+                redirectURL:[NSURL URLWithString:redirectURL]
+               responseType:OIDResponseTypeCode
+       additionalParameters:nil];
+
+  [self createUIWindowForLogin];
+  // performs authentication request
+  [FIRAppDistributionAppDelegatorInterceptor sharedInstance].currentAuthorizationFlow =
+      [OIDAuthState authStateByPresentingAuthorizationRequest:request
+                                     presentingViewController:self.safariHostingViewController
+                                                     callback:^(OIDAuthState *_Nullable authState,
+                                                                NSError *_Nullable error) {
+                                                       self.authState = authState;
+                                                       self.isTesterSignedIn =
+                                                           self.authState ? YES : NO;
+                                                       completion(error);
+                                                     }];
+}
+
+- (UIWindow *)createUIWindowForLogin {
+  // Create an empty window + viewController to host the Safari UI.
+  UIWindow *window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+  window.rootViewController = self.safariHostingViewController;
+
+  // Place it at the highest level within the stack.
+  window.windowLevel = +CGFLOAT_MAX;
+
+  // Run it.
+  [window makeKeyAndVisible];
+
+  return window;
+}
+
+- (void)handleReleasesAPIResponseWithData:data
+                               completion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  // TODO Implement parsing of releases API response
+  completion(nil, nil);
+}
+
+- (void)checkForUpdateWithCompletion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  if (self.isTesterSignedIn) {
+    [self fetchReleases:completion];
+  } else {
+    UIAlertController *alert = [UIAlertController
+        alertControllerWithTitle:@"Enable in-app alerts"
+                         message:@"Sign in with your Firebase App Distribution Google account to "
```

Comment:
It's not clear to me what a "Firebase App Distribution Google account is." I suggest:

`message:@"Sign in with your Google account (the one you use with App Distribution) to..."`